### PR TITLE
Fix checking equipment modifications

### PR DESCRIPTION
**Pull request by flomillot**
Implement the check method when necessary instead of checking data inside the apply method.

Individually check the tabular modifications.

### CODE REVIEW

**Comment by antoinebhs** [comment]
Is there a reason why we didn't do it for these cases ?

https://github.com/gridsuite/network-modification-server/blob/c535de666d98c178be5051be8ff6c00a06301870/src/main/java/org/gridsuite/modification/server/modifications/BatteryModification.java#L78C22-L78C22

https://github.com/gridsuite/network-modification-server/blob/c535de666d98c178be5051be8ff6c00a06301870/src/main/java/org/gridsuite/modification/server/modifications/EquipmentDeletion.java#L38C1-L41C1

https://github.com/gridsuite/network-modification-server/blob/c535de666d98c178be5051be8ff6c00a06301870/src/main/java/org/gridsuite/modification/server/modifications/GroovyScript.java#L37C1-L37C1

**Comment by flomillot** [comment]
> Is there a reason why we didn't do it for these cases ?
> 
> https://github.com/gridsuite/network-modification-server/blob/c535de666d98c178be5051be8ff6c00a06301870/src/main/java/org/gridsuite/modification/server/modifications/BatteryModification.java#L78C22-L78C22
> 
> https://github.com/gridsuite/network-modification-server/blob/c535de666d98c178be5051be8ff6c00a06301870/src/main/java/org/gridsuite/modification/server/modifications/EquipmentDeletion.java#L38C1-L41C1
> 
> https://github.com/gridsuite/network-modification-server/blob/c535de666d98c178be5051be8ff6c00a06301870/src/main/java/org/gridsuite/modification/server/modifications/GroovyScript.java#L37C1-L37C1

Nope, I just did it quickly and maybe forgot some. I will add them.

**Review by klesaulnier — APPROVED** [review]

File: src/main/java/org/gridsuite/modification/server/modifications/TabularModification.java [hunk 1]
```diff
@@ -45,13 +45,15 @@ public void check(Network network) throws NetworkModificationException {
     @Override
     public void apply(Network network, Reporter subReporter) {
         AtomicInteger applicationFailuresCount = new AtomicInteger(0);
-        modificationInfos.getModifications().forEach(modification -> {
+        modificationInfos.getModifications().forEach(modificationInfos -> {
```
Comment (antoinebhs) [inline src/main/java/org/gridsuite/modification/server/modifications/TabularModification.java:48]: We can probably fix the Sonarcube code smell with little effort.

Comment (flomillot) [reply src/main/java/org/gridsuite/modification/server/modifications/TabularModification.java:48]: Didn't see it, done

**Review by antoinebhs — APPROVED** [review]
Code OK

**Review by antoinebhs — APPROVED** [review]

**Pull request merged**